pkg/util/jsonhelper.go

```diff
@@ -17,22 +17,23 @@ package util
 import (
 	"bytes"
 	"encoding/json"
-	"strings"
-
 	"github.com/sirupsen/logrus"
+	"strings"
 )
 
 // MinifyJSON is used to compact the json bytes (e.g. removing whitespaces and new lines character)
-func MinifyJSON(jsonBytes []byte) string {
+func MinifyJSON(jsonBytes []byte, details string) string {
 	buffer := new(bytes.Buffer)
 	if err := json.Compact(buffer, jsonBytes); err != nil {
-		logrus.Warnf("Fail to compact json: %v", err)
-
 		// the uncompleted json will throw an error, i.e. {"foo":"bar"
 		// so instead, switch with the simple parse (e.g. removing new lines)
 		jsonString := string(jsonBytes)
+		// add this log to check which request/response body is not with json format
+		logrus.Warnf("Fail to compact json: %v, source : %s, details: %s", err, jsonString, details)
```

Comment:
Better to indicate if it's from request or response,
e.g. `Fail to compact request json` or `Fail to compact response json`

Reply:
updated @marselsampe